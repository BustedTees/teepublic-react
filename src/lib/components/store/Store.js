import React, { Component } from 'react';

import PropTypes from 'prop-types';
import classnames from 'classnames';

import DesignCollection from '../design_collection/DesignCollection';
import CartButton from '../cart_button/CartButton';
import Pagination from '../pagination/Pagination';
import StoreFilter from '../store_filter/StoreFilter';

import './Store.scss';

const CLASS_ROOT = 'tp-store';

export default class Store extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selectedPage: props.selectedPage || 1,
      selectedAlbumId: parseInt(props.selectedAlbumId, 10) || null,
      selectedProductTypeName: props.selectedProductTypeName
    };
  }

  pageChangeHandler = page => {
    this.setState({ selectedPage: page });
    const { selectedAlbumId, selectedProductTypeName } = this.state;
    this.fetchStore(page, selectedAlbumId, selectedProductTypeName);
  };

  albumChangeHandler = albumId => {
    this.setState({ selectedAlbumId: albumId });
    const { selectedPage, selectedProductTypeName } = this.state;
    this.fetchStore(selectedPage, albumId, selectedProductTypeName);
  };

  productTypeChangeHandler = productTypeName => {
    this.setState({ selectedProductTypeName: productTypeName });
    const { selectedPage, selectedAlbumId } = this.state;
    this.fetchStore(selectedPage, selectedAlbumId, productTypeName);
  };

  fetchStore = (pageNum, albumId, productTypeName) => {
    const { id } = this.props.storeData;

    window.location = this.props.configuration.storeUrl(
      id,
      pageNum,
      albumId,
      productTypeName
    );
  };

  render() {
    const { designs, albums, productTypes } = this.props.storeData._embedded;
    const {
      selectedPage,
      selectedAlbumId,
      selectedProductTypeName
    } = this.state;
    const totalPages = parseInt(this.props.storeData.totalPages, 10);

    const { className } = this.props;
    const classes = classnames(className, CLASS_ROOT, 'teepublic');

    return (
      <div className={classes}>
        <CartButton
          cartUrl={this.props.configuration.cartUrl()}
          onCheckout={cartItems => console.log(cartItems)}
        />

        <StoreFilter
          albums={albums}
          productTypes={productTypes}
          onAlbumChange={this.albumChangeHandler}
          onProductTypeChange={this.productTypeChangeHandler}
          selectedAlbumId={selectedAlbumId}
          selectedProductTypeName={selectedProductTypeName}
        />

        <div className={`${CLASS_ROOT}__collection-wrap`}>
          <StoreFilter
            albums={albums}
            productTypes={productTypes}
            onAlbumChange={this.albumChangeHandler}
            onProductTypeChange={this.productTypeChangeHandler}
            selectedAlbumId={selectedAlbumId}
            selectedProductTypeName={selectedProductTypeName}
            filterStyle="nav"
          />

          <DesignCollection
            designs={designs}
            tileSize="large"
            buyProductLinkBuilder={this.props.configuration.buyProductUrl}
            storeId={this.props.storeData.id}
          />
        </div>

        <Pagination
          currentPage={selectedPage}
          totalPages={totalPages}
          onPageChange={this.pageChangeHandler}
        />
      </div>
    );
  }
}

Store.propTypes = {
  className: PropTypes.string,
  storeData: PropTypes.object.isRequired,
  configuration: PropTypes.shape({
    storeUrl: PropTypes.func.isRequired,
    cartUrl: PropTypes.func.isRequired,
    buyProductUrl: PropTypes.func.isRequired
  }).isRequired,
  selectedAlbumId: PropTypes.number,
  selectedPage: PropTypes.number,
  selectedProductTypeName: PropTypes.string
};
