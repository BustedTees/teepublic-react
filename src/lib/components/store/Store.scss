.tp-store__collection-wrap {
  display: flex;
  margin-top: 25px;
  align-items: flex-start;
}
