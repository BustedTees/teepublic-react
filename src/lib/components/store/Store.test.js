import React from 'react';
import ReactDOM from 'react-dom';
import Store from './Store';

import { ROUTE_CONFIGURATION } from '../../../demo/RouteConfiguration';
import store from '../../../demo/StoreData';

it('Store renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Store
      storeData={store}
      buyProductLinkBuilder={() => {
        '';
      }}
      configuration={ROUTE_CONFIGURATION}
    />,
    div
  );
});
