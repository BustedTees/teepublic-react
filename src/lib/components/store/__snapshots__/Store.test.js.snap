// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="tp-store teepublic"
>
  <div
    className="tp-store__collection-wrap"
  />
</div>
`;
